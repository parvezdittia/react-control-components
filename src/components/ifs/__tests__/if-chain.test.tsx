import React from 'react';
import renderer from 'react-test-renderer';
import { Else, If, IfChain, ElseIf } from '..';

describe('<IfChain /> should throw error', () => {
  it('if the first child is not an If', () => {
    expect(() =>
      renderer.create(
        <IfChain>
          <span>1</span>
          <Else>
            <span>2</span>
          </Else>
        </IfChain>
      )
    ).toThrowErrorMatchingSnapshot();
  });

  it('if the last child is not an Else', () => {
    expect(() =>
      renderer.create(
        <IfChain>
          <If condition={true}>
            <span>1</span>
          </If>
          <span>2</span>
        </IfChain>
      )
    ).toThrowErrorMatchingSnapshot();
  });

  it('if a child between the first and last is not an ElseIf', () => {
    expect(() =>
      renderer.create(
        <IfChain>
          <If condition={true}>
            <span>1</span>
          </If>
          <span>2</span>
          <Else>
            <span>3</span>
          </Else>
        </IfChain>
      )
    ).toThrowErrorMatchingSnapshot();
  });
});

describe('<IfChain />', () => {
  it('should render the first child if truthy', () => {
    const component = renderer.create(
      <IfChain>
        <If condition={true}>
          <span>1</span>
        </If>
        <ElseIf condition={true}>
          <span>2</span>
        </ElseIf>
        <Else>
          <span>3</span>
        </Else>
      </IfChain>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render the first elseif truthy child', () => {
    const component = renderer.create(
      <IfChain>
        <If condition={false}>
          <span>1</span>
        </If>
        <ElseIf condition={true}>
          <span>2</span>
        </ElseIf>
        <Else>
          <span>3</span>
        </Else>
      </IfChain>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render the second elseif truthy child', () => {
    const component = renderer.create(
      <IfChain>
        <If condition={false}>
          <span>1</span>
        </If>
        <ElseIf condition={false}>
          <span>2</span>
        </ElseIf>
        <ElseIf condition={true}>
          <span>3</span>
        </ElseIf>
        <Else>
          <span>4</span>
        </Else>
      </IfChain>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render the last child if no truthy child', () => {
    const component = renderer.create(
      <IfChain>
        <If condition={false}>
          <span>1</span>
        </If>
        <ElseIf condition={false}>
          <span>2</span>
        </ElseIf>
        <Else>
          <span>3</span>
        </Else>
      </IfChain>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should work when only <If /> is child', () => {
    const component = renderer.create(
      <IfChain>
        <If condition={true}>
          <span>1</span>
        </If>
      </IfChain>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should handle nested IfChain', () => {
    const component = renderer.create(
      <IfChain>
        <If condition={false}>
          <span>1</span>
        </If>
        <ElseIf condition={true}>
          <IfChain>
            <If condition={true}>
              <span>4</span>
            </If>
            <ElseIf condition={false}>
              <span>5</span>
            </ElseIf>
            <Else>
              <span>6</span>
            </Else>
          </IfChain>
        </ElseIf>
        <Else>
          <span>3</span>
        </Else>
      </IfChain>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
