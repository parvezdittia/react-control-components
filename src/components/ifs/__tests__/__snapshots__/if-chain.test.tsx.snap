// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IfChain /> should handle nested IfChain 1`] = `
<span>
  4
</span>
`;

exports[`<IfChain /> should render the first child if truthy 1`] = `
<span>
  1
</span>
`;

exports[`<IfChain /> should render the first elseif truthy child 1`] = `
<span>
  2
</span>
`;

exports[`<IfChain /> should render the last child if no truthy child 1`] = `
<span>
  3
</span>
`;

exports[`<IfChain /> should render the second elseif truthy child 1`] = `
<span>
  3
</span>
`;

exports[`<IfChain /> should throw error if a child between the first and last is not an ElseIf 1`] = `"Invalid IfChain"`;

exports[`<IfChain /> should throw error if the first child is not an If 1`] = `"Invalid IfChain"`;

exports[`<IfChain /> should throw error if the last child is not an Else 1`] = `"Invalid IfChain"`;

exports[`<IfChain /> should work when only <If /> is child 1`] = `
<span>
  1
</span>
`;
